btools/core/road/road.py

```diff
@@ -0,0 +1,297 @@
+            face_count += 1
+
+        if prop.generate_right_sidewalk:
+            add_faces_to_map(bm, (bm.edges[face_count].link_faces[0],), FaceMap.SIDEWALK_SIDE)
+            add_faces_to_map(bm, (bm.edges[face_count + 1].link_faces[0],), FaceMap.SIDEWALK)
+        else:
+            add_faces_to_map(bm, (bm.edges[face_count].link_faces[0],), FaceMap.SHOULDER_EXTENSION)
+
+        # Continue to extrude
+        if prop.extrusion_type == "STRAIGHT":
+            cls.extrude_straight(context, prop, bm)
+        else:
+            cls.extrude_curved(context, prop, bm)
+
+        return {"FINISHED"}
+
+    @classmethod
+    @crash_safe
+    def extrude_straight(cls, context, prop, bm):
+        # Add modifiers
+        if not context.object.modifiers:
+            # Array
+            bpy.ops.object.modifier_add(type="ARRAY")
+            modifier = context.object.modifiers["Array"]
+            modifier.show_in_editmode = True
+            modifier.show_on_cage = True
+            modifier.fit_type = "FIT_LENGTH"
+            modifier.fit_length = prop.length
+            modifier.use_merge_vertices = True
+            modifier.relative_offset_displace = [0, 1, 0]
+
+        return {"FINISHED"}
+
+    @classmethod
+    @crash_safe
+    def extrude_curved(cls, context, prop, bm):
+        curve = context.object.children[0]
+
+        # Rotate vertices
+        bmesh.ops.rotate(bm, matrix=Matrix.Rotation(math.radians(90.0), 3, 'Y'), verts=bm.verts)
+
+        # Add modifiers
+        if not context.object.modifiers:
+            # Array
+            bpy.ops.object.modifier_add(type="ARRAY")
+            modifier = context.object.modifiers["Array"]
+            modifier.fit_type = "FIT_CURVE"
+            modifier.use_merge_vertices = True
+            modifier.curve = curve
+            modifier.relative_offset_displace = [0, 1, 0]
+
+            # Curve
+            bpy.ops.object.modifier_add(type="CURVE")
+            modifier = context.object.modifiers["Curve"]
+            modifier.show_in_editmode = True
+            modifier.show_on_cage = True
+            modifier.object = curve
+            modifier.deform_axis = "POS_Y"
+
+        return {"FINISHED"}
+
+    @classmethod
+    @crash_safe
+    def finalize_road(cls, context):
+        if context.active_object is None:
+            return {"FINISHED"}
+        
+        # Apply modifiers
+        bpy.ops.object.modifier_apply(modifier="Array")
+        bpy.ops.object.modifier_apply(modifier="Curve")
+
+        # Remove curve
+        if len(context.active_object.children) > 0 and context.active_object.children[0].type == "CURVE":
+            bpy.data.objects.remove(context.active_object.children[0])
+
+        # Uv calculations
+        bm = bm_from_obj(context.active_object)
+        count = int(context.active_object["VertexCount"])
+        uv_layer = bm.loops.layers.uv.new()
+        sections = math.floor(len(bm.verts) / count)
```

Comment:
Floor division could also work here: `len(bm.verts) // count`. Not sure if its better